server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1572,12 +1566,12 @@ protected DataStoreMetadataUpdateResult updateDataSourceMetadataWithHandle(
 
     if (!startMetadataMatchesExisting) {
       // Not in the desired start state.
-      log.error(
-          "Not updating metadata, existing state[%s] in metadata store doesn't match to the new start state[%s].",
+      return new DataStoreMetadataUpdateResult(true, false, StringUtils.format(
+          "Not updating metadata, existing state[%s] in metadata store doesn't match to the new start state[%s]." +
+              " This usually happens if the input stream is changed within the same supervisor.",
```

Comment:
The resolution for this is to reset the supervisor right? Should we also incorporate that information into this error message?

Reply:
As a newbie to this area, the message is a bit of a mystery. Are we saying that we're using optimistic locking and we just tried to do an update where we expect version X, but found a newer version Y? Or, that the DB transaction failed (perhaps due to locking conflicts) and we discarded the update? Something else?

Also, how does an "input stream change" within "the same supervisor"? Are we saying that the _contents_ of the data in Kafka (say) changed? Or, that the offsets registered in Kafka changed? Or, did I run two supervisors against the same topic?

We also don't say which state we're changing. The state of the supervisor task? Which one?

Is this error a bad thing? Will I lose data? or, will the system continue on and I can ignore the messages? Or, in fact, is this an assertion and, in a properly coded Druid, will never occur?

So, perhaps:

> Attempted to update state for supervisor %s in table %s from %s to %s, but found the state was already at %s. Are you running two supervisors for the same topic?

Don't take the above too literally: I'm making up much of the information.

Reply:
How about 
```
Inconsistent metadata state. This can happen if you update input topic in a spec without changing the supervisor name. Stored state .. new State ...
```

Reply:
Let's see if I can now understand the issue from the error message. The user has a supervisor. They change the Kafka topic name from which that Supervisor reads? When they do that, they have to actually create a new supervisor (i.e. same spec, new name) and remove the old one?

If so, the key wording is "change the input topic name".

Though, this raises a question: why can't we detect that this happened and "do the right thing"? (Though, that is probably a separate PR.)

Reply:
Well doing the right thing could involve some data loss. Thinking a bit more, we could disallow changing the topic name entirely in the same supervisor. If you want to change the topic name, then you have to terminate the existing supervisor and create a new one. Though yes, it could be a separate PR.